File: csharp-playwright-nunit-reqnroll/SauceDemo.UiTests/Pages/CartPage.cs
using Microsoft.Playwright;

namespace SauceDemo.UiTests.Pages;

public sealed class CartPage : BasePage
{
    private const string CartItemNames = "[data-test='inventory-item-name']";
    private const string CheckoutButton = "[data-test='checkout']";

    public CartPage(IPage page) : base(page)
    {
    }

    public async Task<IReadOnlyList<string>> ListItemNamesAsync() =>
        await Page.Locator(CartItemNames).AllTextContentsAsync();

    public async Task<CheckoutInformationPage> StartCheckoutAsync()
    {
        await ClickAsync(CheckoutButton);
        return new CheckoutInformationPage(Page);
    }
}

File: ts-playwright-test/src/pages/cartPage.ts
import type { Locator, Page } from '@playwright/test';
import { BasePage } from './basePage';
import type { TestLogger } from '../support/testLogger';
import { CheckoutInformationPage } from './checkoutInformationPage';

export class CartPage extends BasePage {
  readonly itemNames: Locator;
  private readonly checkoutButton: Locator;

  constructor(page: Page, log: TestLogger) {
    super(page, log);
    this.itemNames = page.getByTestId('inventory-item-name');
    this.checkoutButton = page.getByTestId('checkout');
  }

  async startCheckout(): Promise<CheckoutInformationPage> {
    await this.checkoutButton.click();
    return new CheckoutInformationPage(this.page, this.log);
  }
}
